Automattic/woocommerce-payments
# If COT enabled, use COT query to gather tracking data

Partially fixes #4148 

#### Changes proposed in this Pull Request

The code in `get_mc_order_count` use DB query on posts and post_meta tables to gather data needed for [usage tracking](https://woocommerce.com/usage-tracking/). The code change is to check that if WC has COT and COT enabled, use a query on the order tables, or else fall back to the older query on `posts` and `post_meta`.

#### Testing instructions

1. Have orders placed with multiple currencies
2. Add the following snippet, which logs the tracking data to your `debug.log`
```
add_filter( 'woocommerce_tracker_send_override', '__return_false' );
add_filter( 'woocommerce_tracker_last_send_time', '__return_false' );

add_filter(
	'woocommerce_tracker_data',
	function( $data ) {
		error_log( __FILE__ . ':' . __LINE__ . ":\n" . '$data: ' . print_r( $data, true ) );
		return $data;
	},
	999
);
```
3. Trigger the tracking cron `wp cron event run woocommerce_tracker_send_event` ( This can also be done using WP-Crontrol )
4. Check debug.log for the output
5. The tracking data logged should be the same across various settings, such as
     -  Install older version of WC which is not COT aware
     -  COT enabled with WooCommerce Orders tables as the data store
     -  On WooCommerce ▸ Settings ▸ Advanced ▸ Custom Data Stores, set the data store to use the WordPress posts table as datastore
 6. Place an order and check that the tracking data is updated in all instances.

-------------------


**Post merge**

- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. **Not Applicable**
